cli/azd/internal/appdetect/java.go

```diff
@@ -16,13 +23,121 @@ func (jd *javaDetector) Language() Language {
+			pomFile := filepath.Join(path, entry.Name())
+			project, err := readMavenProject(pomFile)
+			if err != nil {
+				return nil, fmt.Errorf("error reading pom.xml: %w", err)
+			}
+
+			if len(project.Modules) > 0 {
+				// This is a multi-module project, we will capture the analysis, but return nil
+				// to continue recursing
+				jd.rootProjects = append(jd.rootProjects, *project)
+				return nil, nil
+			}
+
+			var currentRoot *mavenProject
+			for _, rootProject := range jd.rootProjects {
+				// we can say that the project is in the root project if the path is under the project
+				if inRoot := strings.HasPrefix(pomFile, rootProject.path); inRoot {
+					currentRoot = &rootProject
+				}
+			}
+
+			_ = currentRoot // use currentRoot here in the analysis
+			result, err := detectDependencies(project, &Project{
 				Language:      Java,
 				Path:          path,
-				DetectionRule: "Inferred by presence of: " + entry.Name(),
-			}, nil
+				DetectionRule: "Inferred by presence of: pom.xml",
+			})
+			if err != nil {
+				return nil, fmt.Errorf("detecting dependencies: %w", err)
+			}
+
+			return result, nil
 		}
 	}
 
 	return nil, nil
 }
+
+// mavenProject represents the top-level structure of a Maven POM file.
+type mavenProject struct {
+	XmlName              xml.Name             `xml:"project"`
+	Parent               parent               `xml:"parent"`
+	Modules              []string             `xml:"modules>module"` // Capture the modules
+	Dependencies         []dependency         `xml:"dependencies>dependency"`
+	DependencyManagement dependencyManagement `xml:"dependencyManagement"`
+	Build                build                `xml:"build"`
+	path                 string
+}
+
+// Parent represents the parent POM if this project is a module.
+type parent struct {
+	GroupId    string `xml:"groupId"`
+	ArtifactId string `xml:"artifactId"`
+	Version    string `xml:"version"`
+}
+
+// Dependency represents a single Maven dependency.
+type dependency struct {
+	GroupId    string `xml:"groupId"`
+	ArtifactId string `xml:"artifactId"`
+	Version    string `xml:"version"`
+	Scope      string `xml:"scope,omitempty"`
+}
+
+// DependencyManagement includes a list of dependencies that are managed.
+type dependencyManagement struct {
+	Dependencies []dependency `xml:"dependencies>dependency"`
+}
+
+// Build represents the build configuration which can contain plugins.
+type build struct {
+	Plugins []plugin `xml:"plugins>plugin"`
+}
+
+// Plugin represents a build plugin.
+type plugin struct {
+	GroupId    string `xml:"groupId"`
+	ArtifactId string `xml:"artifactId"`
+	Version    string `xml:"version"`
+}
+
+func readMavenProject(filePath string) (*mavenProject, error) {
+	bytes, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file: %w", err)
+	}
+
+	var project mavenProject
+	if err := xml.Unmarshal(bytes, &project); err != nil {
+		return nil, fmt.Errorf("error parsing XML: %w", err)
```

Comment:
Slight improvement here:

```suggestion
		return nil, fmt.Errorf("parsing xml: %w", err)
```